networks.go

```diff
@@ -0,0 +1,82 @@
+package glesys
+
+import "context"
+
+// NetworkService provides functions to interact with Networks
+type NetworkService struct {
+	client clientInterface
+}
+
+// Network represents a network
+type NetworkDetails struct {
```

Comment:
For networks we only have a single representation (compared to `Server` and `ServerDetails`) so I think we should be fine with calling this `Network`.